tutorials/documentation_materials/classiq_101/hadamard_test/hadamard_test.ipynb

```diff
@@ -0,0 +1,506 @@
+{
```

Comment:
<p>Good, just add 1 line that states we received the expected result, and why not receiving exactly 0.25.</p><p>Add here the optional exercise: </p><p> </p><p> Run the above example of the Hadamard Test for the QFT and the |0000\rangle statefrom the SDK using 1,000 , 2,000 , 4,000 , 8,000 and 16,000 shots. For each job, calculate the real part of the expectation value using the formula Re{&lt;0|UQFT|0&gt;} = 2P0-1. Plot a graph of the expectation value as a function of the number of shots. Add to the graph the theoretical value. Explain the results, what is the mean and the variance for each execution? Why is that?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/623/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>